girder/api/docs.py

```diff
@@ -18,11 +18,37 @@
 ###############################################################################
 
 import collections
+import functools
 
 
-discovery = set()
-routes = collections.defaultdict(dict)
-models = {}
+# A dict of dicts of lists
+routes = collections.defaultdict(
+    functools.partial(collections.defaultdict, list))
+
+
+def _toRoutePath(resource, route):
+    """
+    Convert a base resource type and list of route components into a
+    Swagger-compatible route path.
+    """
+    # Convert wildcard tokens from :foo form to {foo} form
+    convRoute = [
+        '{{{}}}'.format(token[1:]) if token[0] == ':' else token
```

Comment:
We officially prefer `%` since it handles unicode much better, but haven't squashed out all the old uses of `format`.